CMakeLists.txt

```diff
@@ -16,23 +16,48 @@ option(ENABLE_NESO_TESTS
 include(cmake/Sanitizers.cmake)
 include(cmake/CheckFileList.cmake)
 
+
+
 # ##############################################################################
 # Find dependencies
 # ##############################################################################
-find_package(SYCL REQUIRED)
+
+#USE -DNESO_SYCL_VENDOR to chose implementation at gen time
+set(NESO_SYCL_VENDOR "Default" CACHE STRING "SYCL vendor to look for - required
+if present")
+include(cmake/SYCL.cmake)
+if (NESO_SYCL_VENDOR MATCHES "Default")
+    find_sycl()
+else()
+    find_sycl(${NESO_SYCL_VENDOR})
+endif()
 if(SYCL_FOUND)
-  message(STATUS "Using SYCL implementation ${SYCL_IMPLEMENTATION}")
-  if(hipSYCL_FOUND)
-    set(HIPSYCL_TARGETS "omp")
-    set(SYCL_FLAG NESO_HIPSYCL)
-  elseif(SYCL_IMPLEMENTATION STREQUAL "IntelDPCPP")
-    set(SYCL_FLAG NESO_DPCPP)
-  endif()
+    message(STATUS "Using SYCL implementation ${SYCL_IMPLEMENTATION}")
+    if(NOT SYCL_INTEL)
+        set(HIPSYCL_TARGETS "omp")
+        set(ACPP_TARGETS "omp")
+        set(SYCL_FLAG NESO_HIPSYCL)
+    else()
+        set(SYCL_FLAG NESO_DPCPP)
+    endif()
 endif()
-
-# Use a custom FindNektar++ script to provide an interface target
+#
+# Use a custom FindNektar++ script to vide an interface target
 find_package(Nektar++ REQUIRED)
 
+#Add the NESO-Particles dependencies
+#find_package(NESO-Particles REQUIRED)
+#Alternativly just use as submodule
+option(ENABLE_NESO_PARTICLES_TESTS OFF)
+add_subdirectory(neso-particles)
+#Don't like this add_compile_definition doesn't propogate up
+#so including it here too
+#this won't work if we are using find_package
+#can't just add to target to export 
+#as we might not be using the same compiler
+#that we used to install it
+include(neso-particles/cmake/restrict-keyword.cmake)
```

Comment:
Is the need to include this the reason you've switched from finding NESO-Particles as a package to loading it as a subdirectory?

Might it be possible for NESO-Particles to install the restrict-keyword.cmake file and include it in the exported NESO-ParticlesConfig.cmake?

Reply:
Will this disable the ability to use an external NESO-Particles? i.e. I develop NESO-Particles in a completely separate directory (found as a cmake package).

Reply:
No it should still work as a package - it looks like in dev it copies the cmake files across and I lost that in the changes. I'll fix this now.
I'm not sure what the cleanest way to get it to work in a submodule the add_compile_definitions call won't effect the parent scope.
TBH I think this is overkill anyhow, a header with some ifdefs to cover all the compilers - in practice there are only 3 (clang/icpx, gcc, msvc) of them and that's if we care about windows -  and they all recognise __restrict AFAIK anyhow.